fingerprint/infra/nec-bio-sdk/src/main/java/com/simprints/fingerprint/infra/necsdkimpl/acquisition/template/FingerprintTemplateProviderImpl.kt

```diff
@@ -0,0 +1,109 @@
+
+        // Always require a new image from the scanner using the minimum resolution as we will
+        // process it using secugen image correction
+        log("Acquiring unprocessed image")
+        val unprocessedImage = captureWrapper.acquireUnprocessedImage(
+            Dpi(MIN_CAPTURE_DPI)
+        ).rawUnprocessedImage
+        captureProcessedImageCache.recentlyCapturedImage = unprocessedImage.imageData
+        log("Unprocessed image acquired, processing it")
+        val decodedImage = decodeWsqImage(unprocessedImage)
+        log("Image decoded successfully ${decodedImage.resolution}")
+        log("processing image using secugen image correction")
+        val secugenProcessedImage = processImage(settings, decodedImage)
+        log("quality checking image using nec sdk")
+        val qualityScore = qualityCalculator.getQualityScore(secugenProcessedImage)
+        log("quality score is $qualityScore the threshold is ${settings.qualityThreshold}" )
+        if (qualityScore < settings.qualityThreshold)
+            throw BioSdkException.ImageQualityBelowThresholdException(qualityScore)
+
+        log("extracting template using nec sdk")
+
+
+        return necTemplateExtractor.extract(secugenProcessedImage, qualityScore)
+    }
+
+    private suspend fun readImageDistortionConfiguration() {
+        // if imageDistortionConfiguration not initialized read it from the scanner
+        if (!::imageDistortionConfiguration.isInitialized) {
+            log("Reading image distortion configuration from the scanner")
+            imageDistortionConfiguration =
+                fingerprintCaptureWrapperFactory.captureWrapper.acquireImageDistortionMatrixConfiguration()
+                    .configurationBytes
+            //Todo save the configuration in the datastore for later use to avoid reading it from the scanner every time
+        }
+    }
+
+
+    private fun processImage(
+        settings: FingerprintTemplateAcquisitionSettings,
+        rawImage: FingerprintRawImage
+    ): FingerprintImage {
+
+        val scannerConfig = SecugenImageCorrection.ScannerConfig(
+            imageDistortionConfiguration,
+            settings.processingResolution?.value ?: DEFAULT_RESOLUTION,
+            rawImage.un20SerialNumber,
+            rawImage.brightness
+        )
+        val processedImage =
+            secugenImageCorrection.processRawImage(rawImage.imageBytes, scannerConfig)
+        return FingerprintImage(
+            processedImage.imageBytes,
+            processedImage.width,
+            processedImage.height,
+            processedImage.resolution
+        )
+    }
+
+    private fun decodeWsqImage(unprocessedImage: RawUnprocessedImage): FingerprintRawImage =
+        wsqImageDecoder.decode(unprocessedImage)
+
+    companion object {
+        private const val MIN_CAPTURE_DPI = 500.toShort()
+    }
+}
+
+const val NEC_TEMPLATE_FORMAT = "NEC_1"
+fun log(s: String) {
```

Comment:
At the very least, this function should be private/internal not to pollute the global class path. 
IMO, it is also a bit redundant since just calling Simber is not much more complex (assuming that tag is moved to a constant).